docs/shapes.rst

```diff
@@ -0,0 +1,352 @@
+
+    >>> rectangle_sky = RectangleSkyRegion(center=SkyCoord(42, 43, unit='deg'),
+    ...                                    width=Angle(3, 'deg'), height=Angle(4, 'deg'),
+    ...                                    angle=Angle(5, 'deg'))
+    >>> rectangle_pix = RectanglePixelRegion(center=PixCoord(x=42, y=43),
+    ...                                    width=3, height=4,
+    ...                                    angle=Angle(5, 'deg'))
+
+.. .. _sh-poly:
+..
+.. Polygons
+.. --------
+..
+.. Polygons are the most versatile region, since any region can be approximated
+.. as a polygon.
+..
+.. TODO: explain how polygons are implemented and special polygon methods,
+.. e.g. how to obtain a polygon approximation for any shape.
+.. This is not available yet, for now see `spherical_geometry`_
+.. for spherical polygons and `Shapely`_ for pixel polygons.
+
+.. _sh-wcs:
+
+Transformations
+---------------
+
+In the last two sections, we talked about how for every region shape (e.g.
+circle), there are two classes, one representing "sky regions" and another
+representing "pixel regions" on a given image.
+
+A key feature of the regions package is that, for a given image, more precisely
+a given `~astropy.wcs.WCS` object, it is possible to convert back and forth
+between sky and image regions.
+
+As an example, let's use this :class:`~regions.CircleSkyRegion`, a sky circle region:
+
+.. code-block:: python
+
+    >>> from astropy.coordinates import Angle, SkyCoord
+    >>> from regions import CircleSkyRegion
+
+    >>> center = SkyCoord(50, 10, unit='deg')
+    >>> radius = Angle(30, 'deg')
+    >>> sky_reg = CircleSkyRegion(center, radius)
+
+To convert it to a :class:`~regions.PixelRegion`, call the
+:meth:`~regions.SkyRegion.to_pixel` method:
+
+.. code-block:: python
+
+    >>> pix_reg = sky_reg.to_pixel(wcs)
+    >>> print(pix_reg)
+    CirclePixelRegion
+    center: PixCoord(x=55.35205711214607, y=40.0958313892697)
+    radius: 36.93290808340659
+
+Also to convert a :class:`~regions.PixelRegion` to a
+:class:`~regions.SkyRegion`, call the :meth:`~regions.PixelRegion.to_sky` method:
+
+.. code-block:: python
+
+    >>> sky_reg = pix_reg.to_sky(wcs)
+    >>> print(sky_reg)
+    Region: CircleSkyRegion
+    center: <SkyCoord (ICRS): (ra, dec) in deg
+        (50., 10.)>
+    radius: 30.0 deg
+
+.. _sh-meta:
+
+Meta Data
+---------
+
+A :class:`~regions.Region` has ``meta`` and ``visual`` attributes which
+stores the meta data of the region. Since, this package supports various file
+formats it is necessary to handle the meta attributes supported by them.
+To handle them there are :class:`~regions.RegionMeta` and
+:class:`~regions.RegionVisual` for meta and visual attributes respectively.
+They are subclasses of the python dictionary (`~dict`).
+
+The meta attribute provides additional data about regions such as labels, tags,
+comments, name, etc.. which are used for non-display tasks.
```

Comment:
no double-dot here

Reply:
....unless that's a markdown command I'm not aware of?

Reply:
no special meaning , it's a mistake.